File: app/src/main/java/com/lemuel/lemubit/fingerprinttest/view/ViewAttendance.java
package com.lemuel.lemubit.fingerprinttest.view;

import android.app.Dialog;
import android.app.DialogFragment;
import android.app.TimePickerDialog;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.format.DateFormat;
import android.widget.TimePicker;

import com.lemuel.lemubit.fingerprinttest.R;

import java.util.Calendar;

public class ViewAttendance extends AppCompatActivity {
    //TODO continue from here ViewAttendance
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_view_attendance);

        DialogFragment newFragment = new TimePickerFragment();
        newFragment.show(getFragmentManager(), "timePicker");
    }


    public static class TimePickerFragment extends DialogFragment
            implements TimePickerDialog.OnTimeSetListener {

        @Override
        public Dialog onCreateDialog(Bundle savedInstanceState) {
            // Use the current time as the default values for the picker
            final Calendar c = Calendar.getInstance();
            int hour = c.get(Calendar.HOUR_OF_DAY);
            int minute = c.get(Calendar.MINUTE);

            // Create a new instance of TimePickerDialog and return it
            return new TimePickerDialog(getActivity(), this, hour, minute,
                    DateFormat.is24HourFormat(getActivity()));
        }

        public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
            // Do something with the time chosen by the user
        }
    }
}

File: app/src/main/java/com/lemuel/lemubit/fingerprinttest/view/ViewAttendance.kt
package com.lemuel.lemubit.fingerprinttest.view

import android.app.DatePickerDialog
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.widget.DatePicker
import android.widget.Toast
import com.lemuel.lemubit.fingerprinttest.R
import com.lemuel.lemubit.fingerprinttest.model.DataHelper
import com.lemuel.lemubit.fingerprinttest.recyclerview.ViewAttendanceAdapter
import kotlinx.android.synthetic.main.activity_view_attendance.*
import java.text.SimpleDateFormat
import java.util.*

class ViewAttendance : AppCompatActivity(), DatePickerDialog.OnDateSetListener {
    private var recyclerView: RecyclerView? = null
    private var adapter: ViewAttendanceAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_view_attendance)
        recyclerView = findViewById(R.id.recyclerv_record)

        setUpRecyclerView()

        val c = Calendar.getInstance()
        val year = c.get(Calendar.YEAR)
        val month = c.get(Calendar.MONTH)
        val day = c.get(Calendar.DAY_OF_MONTH)

        btn_filter.setOnClickListener { DatePickerDialog(this@ViewAttendance, this, year, month, day).show() }

        txt_filterdate.setOnClickListener {
            adapter = ViewAttendanceAdapter(DataHelper.getAttendanceRecord(), this)
            recyclerView?.adapter = adapter
            adapter!!.newRecord()
        }

    }

    override fun onDateSet(p0: DatePicker?, p1: Int, p2: Int, p3: Int) {

        val format = formatDate(p1, p2, p3)
        adapter = ViewAttendanceAdapter(DataHelper.getAttendanceRecord(format), this)
        recyclerView?.adapter = adapter
        adapter!!.newRecord()
        Toast.makeText(this@ViewAttendance, format, Toast.LENGTH_SHORT).show()
    }

    private fun formatDate(year: Int, month: Int, day: Int): String {
        val cal = Calendar.getInstance()
        cal.set(year, month, day)
        val format1 = SimpleDateFormat("dd/MM/yyyy", Locale.UK)
        val formatted = format1.format(cal.time)

        return formatted
    }

    private fun setUpRecyclerView() {
        adapter = ViewAttendanceAdapter(DataHelper.getAttendanceRecord(), this)
        recyclerView?.layoutManager = LinearLayoutManager(this)
        recyclerView?.adapter = adapter
        recyclerView?.setHasFixedSize(true)
        recyclerView?.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
    }


}
